src/navigation/delivery/NewDeliveryStore.js

```diff
@@ -21,13 +22,29 @@ const NewDeliveryStore = (props) => {
 
   const [searchQuery, setSearchQuery] = useState('');
   const [storeList, setStoreList] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const stores = useSelector(selectStores)
 
   const {
-    stores,
+    stores: backendStores,
     error,
-    isLoading,
+    isLoading: isLoadingBackendStores,
     refreshStores,
-  } = useFetchAllStores(setStoreList);
+  } = useFetchAllStores({
+    enabled: stores.length === 0,
+  });
+
+  useEffect(() => {
+    if (backendStores) {
+      dispatch(setStores(backendStores));
+    }
+  }, [backendStores, dispatch]);
+
+  useEffect(() => {
+    setIsLoading(isLoadingBackendStores && (
+      !stores || stores.length === 0
+    ));
+  }, [isLoadingBackendStores, stores]);
```

Comment:
Could it be `const isLoading = useMemo(() => isLoadingBackendStores && (!stores || stores.length === 0), [isLoadingBackendStores, stores]);` instead of useState and useEffect?